### DISCUSSION TITLE
Push notification are not supported in this browser

### DISCUSSION THREAD

**Discussion by Datpt2508** [magicbell/community#172] [Questions] [answered]
I used the same code as the example, it works fine on the web but with PWA it returns the message: "Push notification are not supported in this browser". I have put the PWA on the iPad screen. How can I request notification?
![Screenshot 2024-03-18 at 18 52 11](https://github.com/magicbell-io/community/assets/107094515/7c744a87-306d-497d-bd76-c546979451ce)

<img width="570" alt="Screenshot 2024-03-18 at 18 43 59" src="https://github.com/magicbell-io/community/assets/107094515/975e41e1-dbd9-4759-af28-1d1a759c3b9b">

**Comment by unamashana** [comment, marked as answer]
Are you running this on a https URL? 

For anyone else reading, worth mentioning that this is about [webpushtest.com](https://webpushtest.com)

**Reply by Datpt2508** [reply]
i runing on localhost:3000. uhm my bad, i try to use https URL, it working well

**Comment by Datpt2508** [comment]
@unamashana btw i have a question. Does the token magicbell have time to expire ?? I searched the docs but did not see the expiration information of the magicbell token.

**Reply by unamashana** [reply]
@Datpt2508 I need to check the expiration time, but in general, think of it as good for a session. Curious about your question though - what did you have in mind?

**Reply by Datpt2508** [reply]
what I mean is that I want to know the token expiration time so I can handle my next logic. In addition, I want to know how to refresh the token if it expires automatically. Does the magicbell have that feature? i use @magicbell/webpush. and I want to keep my PWA always getting notifications. Thank you so much for support

**Reply by Datpt2508** [reply]
i try to decode my token and get the expiration time, but I don't exactly how long the expiration time works. maybe 12h or 24h??? Do you know the exact expiration time of the token?

<img width="769" alt="Screenshot 2024-03-20 at 10 46 23" src="https://github.com/magicbell-io/community/assets/107094515/6380c30a-6949-4e0d-b3cf-987ffd628947">

**Reply by smeijer** [reply]
Hi @Datpt2508 , the access token to subscribe to push notifications, is a short-lived token. You'll need to request it and directly use it to subscribe. **It expires within an hour.**

It's meant to transfer a session from say your webpage, to a modal (using `window.open`) without exposing sensitive data such as the user hmac in the URL / browser history.

**Reply by Datpt2508** [reply]
i got it, thank you for support
